pkg/providers/equinix.go

```diff
@@ -0,0 +1,121 @@
+package providers
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/packethost/packngo" //requires env var PACKET_AUTH_TOKEN
+)
+
+const (
+	hostPrefix string = "cir"
+)
+
+type equinixProviderConfig struct {
+	ProjectID string   `json:"projectid"` //project id in equinix
+	Facility  []string `json:"facility"`  //datacenters to use
+	Plan      string   `json:"plan"`      //server size
+	OS        string   `json:"os"`        //OS to install
+}
+
+type equinixProvider struct {
+	config equinixProviderConfig
+	client *packngo.Client
+}
+
+func EquinixProviderFactory(providerInfo string, secretData map[string][]byte) (Provider, error) {
+	config := equinixProviderConfig{
+		ProjectID: "f2c6f0a6-4673-464b-878d-cce8dc8d0648",
+		Facility:  []string{"da11", "dc10", "dc13", "ny5", "sv15"},
+		Plan:      "c3.small.x86",
+		OS:        "rocky_8",
+	}
+
+	if configJson, ok := secretData["config"]; ok {
+		if err := json.Unmarshal(configJson, &config); err != nil {
+			return nil, err
+		}
+	}
+
+	client, err := packngo.NewClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &equinixProvider{
+		config: config,
+		client: client,
+	}, nil
+}
+
+func (p *equinixProvider) Acquire() (Resource, error) {
+	resource := Resource{}
+
+	uniqueId := strings.Replace(uuid.New().String(), "-", "", -1)
+	resourceName := fmt.Sprintf("%s-%s", hostPrefix, uniqueId)
+
+	cr := packngo.DeviceCreateRequest{
+		Hostname:  resourceName,
+		Facility:  p.config.Facility,
+		Plan:      p.config.Plan,
+		OS:        p.config.OS,
+		ProjectID: p.config.ProjectID,
+	}
+
+	device, _, err := p.client.Devices.Create(&cr)
+	if err != nil {
+		return resource, err
+	}
+
```

Comment:
we could also test if ssh is available here, I've a branch to do something similar here, but no a PR yet as its waiting for the auth PR to merge
https://github.com/openshift-eng/ofcir/commit/ca441958144c665708e3d4805f2dab93dd23a090

Reply:
I will do this once your utils package merges